src/utils/name_generator.hpp

```diff
@@ -16,20 +16,33 @@
 #define NAME_GENERATOR_HPP_INCLUDED
 
 #include <string>
+#include <exception>
+#include <map>
+
+class name_generator_invalid_exception : public std::exception {
+ public:
+  name_generator_invalid_exception(const char* errMessage):errMessage_(errMessage){}
+  const char* what() const throw() { return errMessage_; }
+ 
+ private:
+  const char* errMessage_;
+};
+
 
 class name_generator {
 public:
+	virtual std::string generate(const std::map<std::string,std::string>& variables) const = 0;
 	virtual std::string generate() const = 0;
-	virtual bool is_valid() const {return true;}
-	virtual ~name_generator() {}
+	name_generator() throw (name_generator_invalid_exception) {};
```

Comment:
There's no need for exception-specifications, and in fact it may be better to avoid them since I believe C++ is phasing them out in later revisions.